Expand the supported version range of sanctuary-def

Avaq opened the pull request:
Sanctuary-Def version 21 updates sanctuary-type-identifiers to
version 3. In order to cooperate with it, we must update
sanctuary-type-identifiers as well.

Fluture only added support for sanctuary-type-identifiers 3 in
12.0.0-beta.6, so as a result of the sanctuary-def version range
expansion, the version range of Fluture must be slightly reduced
from beta 5 at the low edge to beta 6.

That makes this a breaking change.

Avaq committed:
Expand the supported version range of sanctuary-def

Sanctuary-Def version 21 updates sanctuary-type-identifiers to
version 3. In order to cooperate with it, we must update
sanctuary-type-identifiers as well.

Fluture only added support for sanctuary-type-identifiers 3 in
12.0.0-beta.6, so as a result of the sanctuary-def version range
expansion, the version range of Fluture must be slightly reduced
from beta 5 at the low edge to beta 6.

That makes this a breaking change.

Avaq commented:
Although it's technically a breaking change, all we're doing is modifying package meta-data, so users were already able to use fluture-sanctuary-types 5.x in the same way they'll be using 6.x by ignoring the npm warnings. A bit of a bureaucratic update, this.

Avaq commented:
I probably didn't explain enough. The type identifiers bump is *only needed for the tests to pass*. At runtime, it was merely used to identify Fluture types, which supports both specs 2 and 3.

I say this, but it might be worth guarding against the Fluture constructor now being identified as a member of Future after switching to type identifiers 3. Something to consider.

davidchambers commented on package.json:
```diff
@@ -36,19 +36,19 @@
-    "fluture": ">=12.0.0-beta.5 <13.0.0",
-    "sanctuary-def": ">=0.20.0 <0.21.0"
+    "fluture": ">=12.0.0-beta.6 <13.0.0",
+    "sanctuary-def": ">=0.20.0 <0.22.0"
```
Can one use `sanctuary-def@0.20.0` with `sanctuary-type-identifiers@3.0.0`?

Avaq replied:
No, but that's not what would happen. NPM would install two versions of sanctuary-type-identifiers. One used by fluture-sanctuary-types, and one by sanctuary-def.

Since fluture-sanctuary-types doesn't use sanctuary-type-identifiers on any types except those provided by Fluture (with the exception of in the tests), it doesn't matter which version of sanctuary-def is being used.

Avaq committed:
Fix a bug where Par is considered a member of itself

Due to a breaking change in the sanctuary-type-identifiers spec[1],
Fluture adheres to both specs during a transition period.

The only flaw with that approach is that the new
sanctuary-type-identifiers considers constructors to be members of
themselves, and as such, an extra check is needed when identifying
Fluture-exported types using sanctuary-type-identifiers 3.

This extra check can be removed once Fluture drops support for
sanctuary-type-identifiers 2.

[1]: https://github.com/sanctuary-js/sanctuary-type-identifiers/releases/tag/v3.0.0

davidchambers commented on test/index.js:
```diff
@@ -45,6 +46,7 @@ test ('ConcurrentFutureType', () => {
   eq (typeInfo.namespace, 'sanctuary-def');
   eq (typeInfo.name, 'Type');
 
+  eq ($test (Type) (Par), false);
```
:thumbsup:
